sql/catalyst/src/main/java/org/apache/spark/sql/connector/read/SupportsPushDownLimit.java

```diff
@@ -33,4 +34,9 @@
    * Pushes down LIMIT to the data source.
    */
   boolean pushLimit(int limit);
+
+  /**
+   * Pushes down top N to the data source.
+   */
+  boolean pushTopN(SortValue[] orders, int limit);
```

Comment:
There is a little strange that `pushTopN` with the return value of the `Boolean`, How about adding two methods like `pushTopN` and `pushedTopN`, In this way, the responsibilities of each method are cleaner. FYI

Reply:
or probably we should have a new interface `SupportsPushDownTopN`, as Spark can either push down limit, or top n, but not both.